ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -205,6 +241,404 @@ tracerDiffusionSimWithTimeName ntnAddr =
+      sim = diffusionSimulation (toBearerInfo defaultBearerInfo)
+                                diffScript
+                                tracersExtraWithTimeName
+                                tracerDiffusionSimWithTimeName
+
+      events :: [InboundGovernorTrace NtNAddr]
+      events = mapMaybe (\case DiffusionInboundGovernorTrace st -> Just st
+                               _                                -> Nothing
+                        )
+             . Trace.toList
+             . fmap (\(WithTime _ (WithName _ b)) -> b)
+             . withTimeNameTraceEvents
+                @DiffusionTestTrace
+                @NtNAddr
+             . Trace.fromList (MainReturn (Time 0) () [])
+             . fmap (\(t, tid, tl, te) -> SimEvent t tid tl te)
+             . take 125000
+             . traceEvents
+             $ runSimTrace sim
+
+      inboundGovernorTraceMap :: InboundGovernorTrace NtNAddr -> String
+      inboundGovernorTraceMap (TrNewConnection p _)            =
+        "TrNewConnection " ++ show p
+      inboundGovernorTraceMap (TrResponderRestarted _ mpn)         =
+        "TrResponderRestarted " ++ show mpn
+      inboundGovernorTraceMap (TrResponderStartFailure _ mpn se)   =
+        "TrResponderStartFailure " ++ show mpn ++ " " ++ show se
+      inboundGovernorTraceMap (TrResponderErrored _ mpn se)        =
+        "TrResponderErrored " ++ show mpn ++ " " ++ show se
+      inboundGovernorTraceMap (TrResponderStarted _ mpn)           =
+        "TrResponderStarted " ++ show mpn
+      inboundGovernorTraceMap (TrResponderTerminated _ mpn)        =
+        "TrResponderTerminated " ++ show mpn
+      inboundGovernorTraceMap (TrPromotedToWarmRemote _ ora)        =
+        "TrPromotedToWarmRemote " ++ show ora
+      inboundGovernorTraceMap (TrPromotedToHotRemote _)            =
+        "TrPromotedToHotRemote"
+      inboundGovernorTraceMap (TrDemotedToWarmRemote _)            =
+        "TrDemotedToWarmRemote"
+      inboundGovernorTraceMap (TrDemotedToColdRemote _ ora)         =
+        "TrDemotedToColdRemote " ++ show ora
+      inboundGovernorTraceMap (TrWaitIdleRemote _ ora)              =
+        "TrWaitIdleRemote " ++ show ora
+      inboundGovernorTraceMap (TrMuxCleanExit _)                   =
+        "TrMuxCleanExit"
+      inboundGovernorTraceMap (TrMuxErrored _ se)                  =
+        "TrMuxErrored " ++ show se
+      inboundGovernorTraceMap (TrInboundGovernorCounters _)       =
+        "TrInboundGovernorCounters"
+      inboundGovernorTraceMap (TrRemoteState _)                   =
+        "TrRemoteState"
+      inboundGovernorTraceMap (IG.TrUnexpectedlyFalseAssertion _) =
+        "TrUnexpectedlyFalseAssertion"
+      inboundGovernorTraceMap (TrInboundGovernorError se)           =
+        "TrInboundGovernorError " ++ show se
+
+      eventsSeenNames = map inboundGovernorTraceMap events
+
+   in tabulate "inbound governor trace" eventsSeenNames
+      True
```

Comment:
Add a todo & an issue, to actually use `checkCoverage` from `QuickCheck`.  The same for the various other places in this PR.

Reply:
We do use `checkCoverage`, but I think I only added it in a different commit

Reply:
Done!